tests/src/Functional/AccessTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\Tests\og\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Drupal\block_content\Entity\BlockContent;
+use Drupal\block_content\Entity\BlockContentType;
+use Drupal\node\Entity\Node;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Og;
+use Drupal\og\OgGroupAudienceHelper;
+use Drupal\simpletest\ContentTypeCreationTrait;
+
+/**
+ * Tests if the user has access to groups and group content.
+ *
+ * @group og
+ */
+class AccessTest extends BrowserTestBase {
```

Comment:
I'll change it to a KernelTest. I just wanted to validate that the entire chain is working, since access checks are so important. But we can indeed trust that core handles access correctly on their end, and we should only care about our own behaviour.